make error: vfork: resource temporarily unavailable

Hi,

I'm trying to install OpenBLAS 0.2.20 on a node in a local directory over which I have permissions (I don't have root access). I seem to be encountering an error when attempting the build process. Trying `make` with or without any flags is resulting in a long string of errors that look like:
`make[1]: vfork: Resource temporarily unavailable`

I've posted the whole output in a text file. Could anyone give any suggestions on how to troubleshoot the problem? 
[make_error.txt](https://github.com/xianyi/OpenBLAS/files/1445213/make_error.txt)

Many thanks. Tim

EDIT: In case this is useful, here is also the output of a few server parameters.

`uname - or`
2.6.32.54-0.3-default GNU/Linux

`lsb_release -a`
LSB Version:	core-2.0-noarch:core-3.2-noarch:core-4.0-noarch:core-2.0-x86_64:core-3.2-x86_64:core-4.0-x86_64:desktop-4.0-amd64:desktop-4.0-noarch:graphics-2.0-amd64:graphics-2.0-noarch:graphics-3.2-amd64:graphics-3.2-noarch:graphics-4.0-amd64:graphics-4.0-noarch
Distributor ID:	SUSE LINUX
Description:	SUSE Linux Enterprise Server 11 (x86_64)
Release:	11
Codename:	n/a

`cat /etc/*-release`
LSB_VERSION="core-2.0-noarch:core-3.2-noarch:core-4.0-noarch:core-2.0-x86_64:core-3.2-x86_64:core-4.0-x86_64"
SGI Accelerate 1.3, Build 705r10.sles11-1110192111
SGI Foundation Software 2.5, Build 705r10.sles11-1110192111
SGI MPI 1.3, Build 705r10.sles11-1110192111
SGI Performance Suite 1.3, Build 705r10.sles11-1110192111
SGI UPC 1.3, Build 705r10.sles11-1110192111
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 1

`lscpu`:
Architecture:          x86_64
CPU(s):                64
Thread(s) per core:    1
Core(s) per socket:    8
CPU socket(s):         8
NUMA node(s):          8
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 46
Stepping:              6
CPU MHz:               2266.424
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              24576K

Thanks for the comment. I tried setting the `NO_PARALLEL_MAKE` option in `Makefile.rule` but I seem to have stumbled into another problem. I've attached the last few lines below. I'm fairly new to compiling these programs so I may have missed something obvious!

I had also set the prefix options and the gfortran thread safe options.

make[2]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack/trtri'
make[1]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack'
make[1]: Entering directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib'
( cd INSTALL; make )
make[2]: Entering directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib/INSTALL'
gfortran -openmp -fp-model precise -I/ap/vni/imsl/fnl700/lnxsg111e64/include -O2 -frecursive -Wall -m64 -fPIC -c -o lsame.o lsame.f
gfortran: precise: No such file or directory
f951: error: unrecognized command line option "-fp-model"
make[2]: *** [lsame.o] Error 1
make[2]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib/INSTALL'
make[1]: *** [lapack_install] Error 2
make[1]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib'
make: *** [netlib] Error 2

It got a little closer! Using the above (only adding a prefix and MAKE_NB_JOBS=1), it seems that it has defaulted to using ifort, which seems to understand the '-fp-model precise' flag but has the following issue. The last few lines are posted below. Did it fail attempting to compile a fortran code?

...
ifort -openmp -fp-model precise -I/ap/vni/imsl/fnl700/lnxsg111e64/include -O2 -fPIC -c ssytrd_sb2st.F -o ssytrd_sb2st.o
ssytrd_sb2st.F(486): error #5082: Syntax error, found IDENTIFIER 'DEPEND' when expecting one of: UNTIED PRIVATE FIRSTPRIVATE SHARED IF DEFAULT <END-OF-STATEMENT> ;
!$OMP TASK DEPEND(in:WORK(MYID+SHIFT-1))
-----------^
ssytrd_sb2st.F(487): error #5082: Syntax error, found IDENTIFIER 'DEPEND' when expecting one of: ( % . = =>
!$OMP$     DEPEND(in:WORK(MYID-1))
-----------^
ssytrd_sb2st.F(488): error #5082: Syntax error, found IDENTIFIER 'DEPEND' when expecting one of: ( * ) :: , <END-OF-STATEMENT> ; + . - % (/ [ : ] /) . ' ** / ...
!$OMP$     DEPEND(out:WORK(MYID))
-----------^
ssytrd_sb2st.F(488): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: :
!$OMP$     DEPEND(out:WORK(MYID))
---------------------------------^
ssytrd_sb2st.F(497): error #5082: Syntax error, found IDENTIFIER 'DEPEND' when expecting one of: UNTIED PRIVATE FIRSTPRIVATE SHARED IF DEFAULT <END-OF-STATEMENT> ;
!$OMP TASK DEPEND(in:WORK(MYID+SHIFT-1))
-----------^
ssytrd_sb2st.F(498): error #5082: Syntax error, found IDENTIFIER 'DEPEND' when expecting one of: ( % . = =>
!$OMP$     DEPEND(out:WORK(MYID))
-----------^
ssytrd_sb2st.F(497): error #6404: This name does not have a type, and must have an explicit type.   [DEPEND]
!$OMP TASK DEPEND(in:WORK(MYID+SHIFT-1))
-----------^
ssytrd_sb2st.F(497): error #6514: A substring must be of type CHARACTER.   [DEPEND]
!$OMP TASK DEPEND(in:WORK(MYID+SHIFT-1))
-----------^
ssytrd_sb2st.F(497): error #6404: This name does not have a type, and must have an explicit type.   [IN]
!$OMP TASK DEPEND(in:WORK(MYID+SHIFT-1))
------------------^
ssytrd_sb2st.F(498): error #6514: A substring must be of type CHARACTER.   [DEPEND]
!$OMP$     DEPEND(out:WORK(MYID))
-----------^
ssytrd_sb2st.F(498): error #6404: This name does not have a type, and must have an explicit type.   [OUT]
!$OMP$     DEPEND(out:WORK(MYID))
------------------^
compilation aborted for ssytrd_sb2st.F (code 1)
make[2]: *** [ssytrd_sb2st.o] Error 1
make[2]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib/SRC'
make[1]: *** [lapacklib] Error 2
make[1]: Leaving directory `/home/FIa/FIa164/programs/openblas/OpenBLAS-0.2.20_src/lapack-netlib'
make: *** [netlib] Error 2

--- a/driver/others/init.c
+++ b/driver/others/init.c
@@ -90,9 +90,11 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #if defined(BIGNUMA)
 // max number of nodes as defined in numa.h
-// max cpus as defined in sched.h
+// max cpus as defined in most sched.h
+// cannot use CPU_SETSIZE directly as some
+// Linux distributors set it to 4096 
 #define MAX_NODES	128
-#define MAX_CPUS	CPU_SETSIZE
+#define MAX_CPUS	1024
 #else
 #define MAX_NODES	16
 #define MAX_CPUS	256

